examples/sdbus_async/rfcomm-serial-connect.py

```diff
@@ -0,0 +1,114 @@
+
+from sdbus import dbus_method_async_override, sd_bus_open_system
+
+from sdbus_async.bluez.device_api import DeviceInterfaceAsync
+from sdbus_async.bluez.profile_api import (
+    ProfileInterfaceAsync,
+    ProfileManagerInterfaceAsync,
+)
+
+
+def usage() -> None:
+    print(f"Usage: {sys.argv[0]} DEV_ADDR BYTE1 [BYTE2...]")
+    print("\tConnect to serial port service of the Bluetooth RFCOMM device")
+    print("\tspecified with DEV_ADDR and send specified bytes to the device.")
+    print("\t\tDEV_ADDR: e.g. 01:23:45:67:89:AB")
+    print("\t\tBYTE1 [BYTE2...]: e.g. 0x02 0x00 0x01 0x9B")
+    exit(1)
+
+
+if len(sys.argv) < 3:
+    usage()
+
+DEV_ADDR = sys.argv[1]
+DEV_PATH = '/org/bluez/hci0/dev_' + DEV_ADDR.replace(":", "_")
+BYTES_TO_SEND = [int(x, 0) for x in sys.argv[2:]]
+# Serial port service UUID
+SERVICE_UUID = '00001101-0000-1000-8000-00805f9b34fb'
+
+
+class RFCOMMProfileAsync(ProfileInterfaceAsync):
+
+    @dbus_method_async_override()
+    async def release(
+            self,
+    ) -> None:
+        return
+
+    @dbus_method_async_override()
+    async def new_connection(
+            self,
+            device: str,
+            fd: int,
+            fd_properties: Dict[str, Tuple[str, Any]],
+    ) -> None:
+        # Duplicate the file descriptor. The provided file descirptor
+        # will not be usable after new_connection() call end.
+        print(f"Connecting to device: {device}")
+        self.fd = dup(fd)
+
+    @dbus_method_async_override()
+    async def request_disconnection(
+            self,
+            device: str,
+    ) -> None:
+        print(f"Disconnecting device: {device}")
+
+
+async def main() -> None:
+
+    # connect to D-Bus
+    dbus = sd_bus_open_system()
+    profile_manager = ProfileManagerInterfaceAsync()
+    profile_manager._connect('org.bluez', '/org/bluez', bus=dbus)
+    device = DeviceInterfaceAsync()
+    device._connect('org.bluez', DEV_PATH, bus=dbus)
+
+    # Register RFCOMM profile
+    export_object = RFCOMMProfileAsync()
+    export_object.export_to_dbus('/org/test', dbus)
+    await profile_manager.register_profile('/org/test',
+                                           SERVICE_UUID,
+                                           {'Name': ('s', 'foo'),
+                                            'Role': ('s', "client"),
+                                            'Service': ('s', SERVICE_UUID),
+                                            'Channel': ('i', 0),
+                                            'AutoConnect': ('b', True),
+                                            })
+
+    # Connect to serial port service
+    try:
+        await device.connect_profile(SERVICE_UUID)
+    except NotImplementedError as e:
+        print(e)
+        return
+    except Exception as e:
+        print(e)
+        print(f"failed to connect: {e}")
+        return
+
+    # Send and receive data
+    fp = fdopen(export_object.fd, mode='r+b', buffering=0, newline=None)
```

Comment:
I guess there is some other device that will read and write to us?

Reply:
The example does both write and read. The file descriptor is opened with "r+" mode, not "r", so we can do both write and read to the descriptor. In actual application, it would be the better to have multiple threads dedicated for each of read and write but I think it is beyond scope of this example.

Reply:
What I am afraid of is that it does not actually send anything over Bluetooth.

I think using Wireshark to monitor the bluetooth would show if the Bluetooth is actually used.

Reply:
Okay, it sounds a fun to check with Wireshark :)

I ran Wireshark Bluetooth capture and then ran the example script for my Lego Mindstorms NXT Intelligent Brick. As script arguments, I specified device address "00:16:53:07:D9:86" and 4 bytes data to send "0x02 0x00 0x01 0x9B", which requests device information to the brick. The script printed byte string send back from the brick:

```
Received bytes:  b'!\x00\x02\x9b\x00NXT071\x00\x00\x00\x00\x00\x00\x00\x00\x00\x00\x16S\x07\xd9\x86\x00\x00\x00\x00\x00d:\x01\x00'
```

In this received bytes, the string "NXT071" is the name of the device.  "\x00\x16S\x07\xd9\x86" is the device address.

From the captured packets, I found two packets:
[rfcomm-serial-connect-wireshark-packets.txt](https://github.com/python-sdbus/python-sdbus-bluez/files/10145937/rfcomm-serial-connect-wireshark-packets.txt)
The first packet dump shows the 4 bytes sent: "02 00 01 9b". And the second packet dump shows the exactly same bytes that the script printed.  These prove that the script communicated with the Bluetooth device for real.

Reply:
Great! Thank you for verifying it.